### PR TITLE
[16.0][MIG] account_statement_import_online_gocardless

### PR DESCRIPTION
**Pull request by njeudy**
Using initial forgeflow migration PR, and add some update

### CODE REVIEW

**Comment by njeudy** [comment]
continue the work for https://github.com/OCA/bank-statement-import/pull/641

**Comment by njeudy** [comment]
@pedrobaeza do i need to add a timeout to request ?

**Comment by pedrobaeza** [comment]
Yes, please. There are missing commits that has landed in 15.0 branch later than the original PR. Please cherry-pick them. You need to reorder the commits into a logical sequence (the migration commits to be the last), and please remember other times to put correctly the PR title and don't let the default proposed one.

**Comment by njeudy** [comment]
@pedrobaeza I push all, and decide to set timeout to 10 min because of the first import that could or for some big bank statement file .. what do you think ? I can add a resconfig parameter to if needed ..

**Comment by pedrobaeza** [comment]
Thanks. That huge timeout is not really needed. Take into account that the timeout is just for the HTTP request, not for all process. This takes at maximum seconds, and although you ask for whole years, the requests are split by the statement interval set in the journal. And I think this timeout is only for getting in contact, not for finishing whole request.

I think 5 seconds is the ideal timeout.

**Comment by njeudy** [comment]
ok I change to 5 sec and will see if any issue with this .. Thanks for your help ..

**Comment by njeudy** [comment]
@pedrobaeza I finish the migration, rebase all commits and test it with real sync (gocardlesse <-> qonto) and all is ok for me.
Now, some test need to be done as I see, but don't know how to test it because of api call ..

File: account_statement_import_online_gocardless/tests/test_account_statement_import_online_gocardless.py [hunk 1]
```diff
@@ -80,7 +80,6 @@ def setUpClass(cls):
 
     def test_mocked_gocardless(self):
         vals = {
-            "provider_ids": self.provider.ids,
```
Comment (pedrobaeza) [inline account_statement_import_online_gocardless/tests/test_account_statement_import_online_gocardless.py]: Why removing this?

**Review by pedrobaeza — COMMENTED** [review]
I don't see the timeouts applied. Please squash latest commit into the migration one.

**Comment by njeudy** [comment]
I squash all commit. Timeout are set in request params dans defined by default on top of py file ...

**Comment by njeudy** [comment]
@alexis-via can you check this, I think it can replace myponto :) I need help to add test on this

File: account_statement_import_online_gocardless/tests/test_account_statement_import_online_gocardless.py (hunk 1 above)
Comment (pedrobaeza) [reply account_statement_import_online_gocardless/tests/test_account_statement_import_online_gocardless.py]: Seeing the tests, now I see why you removed it. I was just doing the question for knowing it. Anyway, I see that you were the one removing that field:

https://github.com/OCA/bank-statement-import/commit/bd10c2f59e4c37cc9b9c400bd3eab3a192b257c0#diff-724a1bdf113d58c503e61bbc4e54e4416dc30cc2246214f19bbe2e521539eadeL24

Please remove it again.

**Comment by pedrobaeza** [comment]
@njeudy I'm not able to push to your branch. Please remove again the line `"provider_ids": self.provider.ids,` and this is good to be merged.

**Comment by njeudy** [comment]
@pedrobaeza done, thanks for your time !

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**